### PR TITLE
Fix removed custom headers still exist issue in zendesk webhook after deploment

### PR DESCRIPTION
**Pull request by cl8au**
👋  Salto team, as I dive into the Salto code base for the first time, I'm making an effort to thoroughly understand the existing logic and context. If I have overlooked any details or misinterpreted any part of the code, I would greatly appreciate your input. Please don't hesitate to offer feedback. Thank you for your assistance.

There is an alternative PR https://github.com/salto-io/salto/pull/5520 which just updating the HTTP method from `PATCH` to `PUT` for modification behaviour of Zendesk webhook in order to address this issue. The reason I've created two PRs is that I am not sure the reason of using `PATCH` of webhook mortification behaviour in deployment.

# Description of this PR
We've identified a minor issue during the deployment of webhooks with custom headers from the source environment to the target environment, specifically related to the `PATCH` behaviour. When custom headers are removed from a webhook in the source environment, these changes are not being reflected in the target environment; the removed headers persist in the target webhook.

## Expected behaviour
| Source Webhook Custom Headers   |    Change Behaviour     |  Target Webhook Custom Headers |
|----------|:-------------:|------:|
| h1:v1 |  delete | deleted |
| h2:v2 |  change value from v2 to v2_changed | h2:v2_changed |
| not presented |  add h3:v3 | h3:v3 |
| h4:v4 | no change, value is still v4 | h4:v4 |

## Existing behaviour
| Source Webhook Custom Headers   |    Change Behaviour     |  Target Webhook Custom Headers |
|----------|:-------------:|------:|
| h1:v1 |  delete | h1:v1 |
| h2:v2 |  change value from v2 to v2_changed | h2:v2_changed |
| not presented |  add h3:v3 | h3:v3 |
| h4:v4 | no change, value is still v4 | h4:v4 |

In [Zendesk Webhook API doc](https://developer.zendesk.com/api-reference/webhooks/webhooks-api/webhooks/), I believe it missed explicit description of how to delete custom header. By using browser inspector, I believe setting `null` value of a header is the way to delete this header.
Screenshot 2024-02-28 at 19 58 57

## Changes
- Set `null` for the custom headers which are removed for Zendesk webhook update

## Screenshots
Sandbox webhook removed `h1` and saved change
Screenshot 2024-02-28 at 19 26 58

Salto change set view shows `h1` should be removed
Screenshot 2024-02-28 at 19 27 50

Production webhook still contains `h1` which should be removed after deployment
Screenshot 2024-02-28 at 19 28 59

---

_Additional context for reviewer_

I was able to run the jest tests locally but could not figure out how to run Salto locally. Therefore, my assumption of the fix is based on the jest test results. Would you please help me test it in your staging if you feel this PR fix is reasonable? Many thanks for the help again.

Screenshot 2024-02-29 at 10 04 28

---
_Release Notes_: 
zendesk:
* Fix removed custom headers still exist in zendesk webhook after deployment

---
_User Notifications_: 
None

### CODE REVIEW

**Review by edenhassid — COMMENTED** [review]
Thanks for the PR this looks great! 🔥 I added one small comment 😄

File: packages/zendesk-adapter/src/filters/webhook.ts [hunk 1]
```diff
@@ -53,6 +53,15 @@ const filterCreator: FilterCreator = ({ config, client }) => ({
         } else if (instance.value.authentication === undefined) {
           instance.value.authentication = null
         }
+
+        if (!_.isEqual(clonedChange.data.before.value.custom_headers, clonedChange.data.after.value.custom_headers)) {
+          // remove any custom headers which no longer needed in after webhook.
+          _.forEach(_.keys(clonedChange.data.before.value.custom_headers), key => {
+            if (!_.has(instance.value.custom_headers, key)) {
+              instance.value.custom_headers[key] = null
```
[the remark this answers is not in the document]
Comment (cl8au) [reply packages/zendesk-adapter/src/filters/webhook.ts:64]: @edenhassid Thanks for the suggestions and I will apply the suggestion in this PR

Just like to double check the logic with you first :) There is `!_.isNil(instance.value.custom_headers)` on https://github.com/salto-io/salto/pull/5516/files#diff-b6ee0a26d276a0c556436600109e0d908fca47268056f1569d6b4186eb50a40eR60 which checks whether `custom_headers` of instance.value is null or undefined or not.

Does it mean with `_.set(instance.value, ['custom_headers', key], null)`, I should be able to remove this check in this case?

Again, many thanks for the review and help :)

[the remark this answers is not in the document]
Comment (cl8au) [reply packages/zendesk-adapter/src/filters/webhook.ts:64]: I think I get what you mean, so what you mean here is that if `custom_headers` becomes `undefined`, you'd like to transform them to be 
```
{
  h2: null,
  h3: null,
  h4: null
}
```

[the remark this answers is not in the document]
Comment (cl8au) [reply packages/zendesk-adapter/src/filters/webhook.ts]: @edenhassid I've applied fixup commit https://github.com/salto-io/salto/pull/5516/commits/ee4298b938fab35db1b969e4ff64a45401f18f9a. Please have a look again when you have time :)

[the remark this answers is not in the document]
Comment (edenhassid) [reply packages/zendesk-adapter/src/filters/webhook.ts]: You can find the steps to run Salto on your local machine in the user guide at https://github.com/salto-io/salto/blob/main/docs/user_guide.md. If you run into any problems, feel free to reach out :). Also, note that after executing `yarn build`, the Salto executable will be available at `/salto/packages/cli/bin/salto` for you to use.

**Review by edenhassid — APPROVED** [review]
LGTM! 💯 🏆

**Comment by cl8au** [comment]
> LGTM! 💯 🏆
### Before deployment, this is in existing production webhook
Screenshot 2024-03-01 at 15 09 00

### I made changes locally to remove 1 and start the new deployment from CLI
Screenshot 2024-03-01 at 15 07 48
Screenshot 2024-03-01 at 15 09 17

### After deployment, h1 has been removed
Screenshot 2024-03-01 at 15 09 27

@edenhassid I believe this test result shows the fix works

**Pull request merged**